src/index.js

```diff
@@ -29,70 +29,91 @@ function App() {
-            }
-            currentSort[colToSort] = direction;
+        // goal is to ony 1 key-value pair at any time, so the render action have only 1 thing to recognize even without a click trigger
+        // 1. if exists -->> flip it
+        // 2. if dont exists --> destroy everything in it, create this key
+        if(sortConfig[colToSort]){
+            direction = sortConfig[colToSort] === 'asc'? 'des' : 'asc';
+        }
+        else{
+            direction = 'asc';
+        }
+        newSortConfig[colToSort] = direction;
+        setSortConfig(newSortConfig);
+    }
 
-            setSortConfig(currentSort);
+    const renderList = () => { // or this can just be a simple render function, that checks for sort btw every time
+        const currentTodos = [...todos];
+        // not sure if this is a good practice t o use Object.entries here, but now there is always only 1 pair of key:value, this should be good functionally?
+        const currentSortConfig = Object.entries(sortConfig);
+        let colToSort = null;
+        let direction = null;
+
+        if(currentSortConfig.length) { 
+            colToSort = currentSortConfig[0][0];
+            direction = currentSortConfig[0][1];
+        }
 
-            if(currentTodos) {
-                if(colToSort === 'date') {
-                    currentTodos.sort((a,b) => {
-                        if(+(new Date(a.due)) < +(new Date(b.due))) return direction === 'asc'? -1 : 1;
-                        if(+(new Date(a.due)) > +(new Date(b.due))) return direction === 'asc'? 1 : -1;
-                        return 0;
-                    });
+        if(currentTodos){
+            if(colToSort === 'date'){
+                if(direction === 'asc'){
+                    currentTodos.sort((a, b) => {
+                        return new Date(a.due) - new Date(b.due);
+                    })
                 }
-                if(colToSort === 'task') {
-                    currentTodos.sort((a,b) => {
-                        if(a.name.toLowerCase() > b.name.toLowerCase()) return direction === 'asc'? -1 : 1;
-                        if(a.name.toLowerCase() < b.name.toLowerCase()) return direction === 'asc'? 1 : -1;
-                        return 0;
-                    });
+                else {
+                    currentTodos.sort((a, b) => {
+                        return new Date(b.due) - new Date(a.due);
+                    })
                 }
-                setToDos(currentTodos);
             }
+            
+            if(colToSort === 'task'){
+                currentTodos.sort((a,b) => {
+                    if(a.name.toLowerCase() > b.name.toLowerCase()){
+                        return direction === 'asc'? -1 : 1;
+                    } 
+                    if(a.name.toLowerCase() < b.name.toLowerCase()){
+                        return direction === 'asc'? 1 : -1;
+                    } 
+                });
```

Comment:
This is fine for now but I feel like there's better way to do this.

Reply:
localeCompare?